platforms/software/testing-base-infrastructure/src/main/java/org/gradle/api/internal/tasks/testing/DirectoryBasedTestDefinition.java

```diff
@@ -52,7 +52,10 @@ public String getDisplayName() {
 
     @Override
     public boolean matches(TestSelectionMatcher matcher) {
-        // TODO: Implement filtering
+        // The only place this method is called is from PatternMatchTestDefinitionProcessor,
+        // where we want to let all candidate test definition dirs that have already
+        // been filtered in the Test task itself through to the workers.
+        // So we'll just let everything through here.
```

Comment:
This is Gate (B) filtering from the Spec, which isn't necessary or beneficial.

It probably means we could move `matches()` out of `TestDefinition` entirely, but I kept it simple for now.